Kernel/Net/IPv4Socket.cpp

```diff
@@ -45,11 +46,12 @@
 
 namespace Kernel {
 
+static Singleton<Lockable<HashTable<IPv4Socket*>>, []() {
+    return new Lockable<HashTable<IPv4Socket*>>;
+}> s_table;
+
```

Comment:
This feels like macro territory.

Reply:
What if the code in the initialization function needs to be a bit more complex? Or are you purely referring to the definition?